### PR TITLE
build separate kubectl instead of renaming openshift

### PR DESCRIPTION
**Pull request by deads2k**
Alright, let's start small. Build the actual `kubectl` instead of renaming openshift.

### CODE REVIEW

**Comment by smarterclayton** [comment]
And this kubectl wouldn't work with openshift objects for get, describe, apply, or the set commands?  Or would it.

**Comment by smarterclayton** [comment]
One that doesn't isn't interesting to me.

**Comment by deads2k** [comment]
> And this kubectl wouldn't work with openshift objects for get, describe, apply, or the set commands? Or would it.

So we don't want `kubectl`, we want a fork of `kubectl`?  That's a wrinkle I didn't expect.  Seems like it would produce some annoying service questions, "which kubectl are you really running.  You can't tell using `kubectl version` (it lies), try ....?".

@liggitt You'll carry a fork of `kubectl` with you as part of `oc` (it would need a complete factory) or you'd want to resolve it a different way?

**Comment by deads2k** [comment]
@stevekuznetsov the ci/openshift-jenkins/extended_conformance_install_update error doesn't look like one I caused.  Something about a tagged 3.7?

**Comment by deads2k** [comment]
> And this kubectl wouldn't work with openshift objects for get, describe, apply, or the set commands? Or would it.

Made it do that.  I can't say that I'm particularly pleased with the notion, but it does that.

**Comment by deads2k** [comment]
> EDIT: i.e. if we don't support openshift objects in the legacy path, we should just stop building kubectl completely.

That would make me much happier.  Is that an option on the table?

File: hack/lib/constants.sh [hunk 1]
```diff
@@ -76,7 +77,6 @@ readonly OPENSHIFT_BINARY_SYMLINKS=(
   osc
   oadm
   osadm
-  kubectl
```
Comment (liggitt) [inline hack/lib/constants.sh]: what does OC_BINARY_COPY do? looks like it already symlinks oc to kubectl

Comment (deads2k) [reply hack/lib/constants.sh]: > what does OC_BINARY_COPY do? looks like it already symlinks oc to kubectl

I wasn't going to remove that one yet.  This is practice for finding places to adjust as I remove `kube-apiserver`, `kubelet`, etc in bulk.

**Comment by deads2k** [comment]
> EDIT: i.e. if we don't support openshift objects in the legacy path, we should just stop building kubectl completely.
> That would make me much happier. Is that an option on the table?

Removing per discussion.

File: test/cmd/basicresources.sh [hunk 2]
```diff
@@ -154,7 +146,7 @@ os::cmd::expect_success 'oc get nodes'
   # subshell so we can unset kubeconfig
```
Comment (liggitt) [inline test/cmd/basicresources.sh]: can delete this whole subshell stanza

File: origin.spec [hunk 3]
```diff
@@ -290,7 +290,7 @@ PLATFORM="$(go env GOHOSTOS)/$(go env GOHOSTARCH)"
 install -d %{buildroot}%{_bindir}
 
 # Install linux components
-for bin in oc openshift dockerregistry kubefed template-service-broker
+for bin in oc openshift dockerregistry kubefed template-service-broker kubectl
```
Comment (liggitt) [inline origin.spec]: drop this, right?

**Comment by deads2k** [comment]
@sdodson how would I debug:

```
+ echo 'openshift-ansible openshift-ansible-callback-plugins openshift-ansible-docs openshift-ansible-filter-plugins openshift-ansible-lookup-plugins openshift-ansible-playbooks openshift-ansible-roles'
++ cat OPENSHIFT_ANSIBLE_BUILT_VERSION
+ sudo python sjb/hack/determine_install_upgrade_version.py atomic-openshift-utils-3.8.1-1.git.0.a172e47.el7.noarch --dependency_branch master
[ERROR] Can not determine install and upgrade version for the `atomic-openshift-utils` package
++ export status=FAILURE
++ status=FAILURE
+ set +o xtrace
########## FINISHED STAGE: FAILURE: INSTALL ANSIBLE ATOMIC-OPENSHIFT-UTILS [00h 01m 18s] ##########
```
from https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/17264/test_pull_request_origin_extended_conformance_install_update/9286/

File: test/cmd/basicresources.sh (hunk 2 above)
Comment (deads2k) [reply test/cmd/basicresources.sh]: > can delete this whole subshell stanza

You don't think the point of the test was to see if an explicit kubeconfig worked?  I thought the `kubectl` aspect was ancillary.

Comment (liggitt) [reply test/cmd/basicresources.sh]: yeah, sorry... misread the change and though you removed line 157 completely

**Comment by deads2k** [comment]
@smarterclayton @liggitt looks like e2e tests look for kubectl.  How would we like to get it?

**Comment by stevekuznetsov** [comment]
@deads2k just so we know -- what are the impacts on a) build time and b) size?

**Comment by deads2k** [comment]
> @deads2k just so we know -- what are the impacts on a) build time and b) size?

For this?  Nothing, since it is a straight removal.

In general, RPM subpackages are supposed to limit the size impact.  The time impact: I think we just have to bite the bullet.

**Pull request closed without merging**